### PR TITLE
Palinilap

### PR DESCRIPTION
**Pull request by SansPapyrus683**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

think java is possible but lambdas in that language are absolute ass

### CODE REVIEW

File: solutions/coi-16-palinilap.mdx [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+	}
+	// these two combined give the bad array stated in the analysis
+	vector<long long> bad1 = range_res(len, removal1);
+	vector<long long> bad2 = range_res(len, removal2);
+	std::reverse(bad2.begin(), bad2.end());
+	
+	long long max_weight = init_amt;
+	for (int i = 0; i < len; i++) {
+		long long max_inc = 0;
+		for (const auto& [_, i] : good[i]) {
```
Comment (ryanchou-dev) [inline solutions/coi-16-palinilap.mdx:269]: weren't you against auto

File: solutions/coi-16-palinilap.mdx [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+		pair<int, char> rep = {c + raw_pal + 1, str[c - raw_pal - 1]};
+		check = [&](int d) { return is_pal_rep(c - d, c + d, rep); };
+		int rep_pal = last_true(raw_pal + 1, most, check);
+		if (rep_pal != -1) {
+			good[rep.first][rep.second] += rep_pal - raw_pal;
+			good[c - raw_pal - 1][str[c + raw_pal + 1]] += rep_pal - raw_pal;
+		}
+
+		// and then the even-llength palindromes
```
Comment (ryanchou-dev) [inline solutions/coi-16-palinilap.mdx:237]:
```suggestion
		// and then the even-length palindromes
```

File: solutions/coi-16-palinilap.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/coi-16-palinilap.mdx:269]: there's no way to do this w/o auto

File: solutions/coi-16-palinilap.mdx [hunk 3]
```diff
@@ -0,0 +1,280 @@
+---
+id: coi-16-palinilap
+source: COI 2016
+title: Palinilap
+author: Kevin Sheng
+---
+
+[Official Analysis (C++)](https://github.com/mostafa-saad/MyCompetitiveProgramming/blob/master/Olympiad/COI/official/2016)
```
Comment (dongliu0426) [inline solutions/coi-16-palinilap.mdx:8]: link https://github.com/mostafa-saad/MyCompetitiveProgramming/blob/master/Olympiad/COI/official/2016/solutions.pdf as well?

Comment (SansPapyrus683) [reply solutions/coi-16-palinilap.mdx:8]: if you click my link solutions.pdf is right there?

**Review by dongliu0426 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**